test/arrays/bradc/noelemtype.future

```diff
@@ -0,0 +1 @@
+#21572: Require arrays always specify an element type
```

Comment:
When reviewing the test changes here, I originally wrote:

> Rather than turning this test into a future, I'd suggest either:
> * replace the lack of an element type with a `?` ("the new way to write an array without an element type")
> * or capture the current error messages when an element type is omitted into the .good
> (where the second is more similar to what this test has been traditionally doing; but if you already have a bunch of tests locking that in, maybe the first would be more interesting).

but then looking at the .bad file, I'm not clear why the compiler is inferring `A` to be an array of domains—is that clear to you?

(in any case, I wouldn't make this a future, or at least not a future that points to #21572, since this PR is intended to resolve that issue.  I.e., assuming we require an element type in #21572 as seems to be the current trend, we wouldn't expect the `.good` file here to be generated would we?)